myapp/app/controllers/application_controller.rb

```diff
@@ -18,4 +19,12 @@ def record_not_found(e = nil)
     end
     redirect_to root_path, danger: t('error.messages.record_not_found')
   end
+
+  def maintenance_mode_on!
+    mainte_flg = Maintenance.exists?(status: 1)
+
+    return unless mainte_flg == true
+
+    redirect_to maintenance_path
```

Comment:
・boolean を true/false と比較するのはナンセンスなので不要です
・1度しか利用しない値を変数化する必要はないです

ので
```suggestion
    return unless Maintenance.exists?(status: 1)

    redirect_to maintenance_path
```

とか、なんなら
```suggestion
    return redirect_to maintenance_path if Maintenance.exists?(status: 1)
```
のように1行でも良いと思います

Reply:
あ、
>1度しか利用しない値を変数化する必要はないです

そういった場合でも変数化したほうが良い場合もないわけではなくて、条件が複雑だったりしてパッと見てわからない場合は
```ruby
mainte_flg = メンテナンスモードを決定するための、非常に複雑で長くチェーンしたメソッド呼び出し
```
みたいにすると「よくわからんがとりあえずメンテフラグなんだな」と理解できるので、そこが修正や解読の主題でない場合にスキップできるので助かります